src/spikeinterface/core/generate.py

```diff
@@ -317,6 +318,93 @@ def generate_snippets(
 
 
 ## spiketrain zone ##
+def synthesize_poisson_spike_vector(
+    num_units=20,
+    sampling_frequency=30000.0,
+    duration=60.0,
+    refractory_period_ms=4.0,
+    firing_rates=3.0,
+    seed=0,
+):
+    """
+    Generate random spike frames for neuronal units based on a Poisson process.
+
+    This function simulates the spike frames of a number of neuronal units, each firing according to
+    a Poisson process. The spike times are discretized to align with a given sampling frequency.
+
+    Parameters
+    ----------
+    num_units : int, default: 20
+        Number of neuronal units to simulate
+    sampling_frequency : float, default: 30000.0
+        Sampling frequency in Hz
+    duration : float, default: 60.0
+        Duration of the simulation in seconds
+    refractory_period_ms : float, default: 4.0
+        Refractory period between spikes in milliseconds
+    firing_rates : float or array_like, default: 3.0
+        Firing rate(s) in Hz. Can be a single value for all units or an array of firing rates with
+        each element being the firing rate for one unit
+    seed : int, default: 0
+        Seed for random number generator
+
+    Returns
+    -------
+    spike_frames : ndarray
+        1D array of spike frames.
+    unit_indices : ndarray
+        1D array of unit indices corresponding to each spike.
+
+    Notes
+    -----
+    - The function uses a geometric distribution to simulate the discrete inter-spike intervals (in frames).
+    This is the discrete equivalent of an exponential distribution which is the distribution of
+    the inter-spike intervals (in times) when the spikes are generated by a Poisson process.
+    - The refractory period is implemented by adding a fixed number of frames to the inter-spike intervals.
+    """
+
+    rng = np.random.default_rng(seed=seed)
+
+    if np.isscalar(firing_rates):
+        firing_rates = np.full(num_units, firing_rates, dtype="float64")
+
+    refractory_period_seconds = refractory_period_ms / 1000.0
+    refactory_period_frames = int(refractory_period_seconds * sampling_frequency)
+
+    # Equivalence between exponential and geometric distributions scaled to the discrete mean
+    geometric_p = 1 - np.exp(-firing_rates / sampling_frequency)
+
+    # We generate as many spikes as the mean plus two std to be sure we have enough
+    max_frames = duration * sampling_frequency
+    max_p = geometric_p.max()
+    num_spikes_expected = ceil(max_frames * max_p)
+    num_spikes_std = int(np.sqrt(num_spikes_expected * (1 - max_p)))
+    num_spikes_max = num_spikes_expected + 2 * num_spikes_std
+
+    # Generate inter spike frames, add the refactory samples and accumulate for sorted spike frames
+    inter_spike_frames = rng.geometric(p=geometric_p[:, np.newaxis], size=(num_units, num_spikes_max))
```

Comment:
Why use this complex stuff rather than a simple `refractory_period + np.random.exponential(scale=beta, size=...)`?
Is this faster / more memory efficient?

+ only 2*std will fail around 2.5% of the time. I think 4*std is best :)

Reply:
> There is a memory spikes coverage trade-off here.
I am not sure if you understood me. The longer the cutout (the one that I am making 2 * stds) the more extra memory I have to allocate and then drop. Maybe I am not understanding what you mean by negligible?

I am not sure what you mean by bias, it is surely not in a formal sense, right (like a biased estimator). Can you be more specific about what statistical effect you are talking about? 

> To me it sounds statistically easier to think about it that way, and the code is easier to read (but it might be just me)
You might be right about it. If there are no memory-costs or if there is something wrong statistically I might change to this. I don't think that you can use float allocated space to allocate integers though so I don't know if you avoid memory costs and the argument will have to rely in something being wrong statistically.